### PR TITLE
Intel offload

### PR DESCRIPTION
**Pull request by jmohdyusof**
Pull request for initial pass of Intel offloaded dense functionality

### CODE REVIEW

**Review by suemni — APPROVED** [review]

File: tests/C-tests/threshold_matrix_typed.c [hunk 1]
```diff
@@ -75,7 +77,11 @@ int TYPED_FUNC(
     A = bml_random_matrix(matrix_type, matrix_precision, N, M, distrib_mode);
     REAL_T scale_factor = 1.e-2;
     bml_scale_inplace(&scale_factor, A);
-    bml_print_bml_matrix(A, 0, max_row, 0, max_col);
+    if (bml_getMyRank() == 0)
+    {
+        printf("A scaled by %f \n", scale_factor);
+        bml_print_bml_matrix(A, 0, max_row, 0, max_col);
```
Comment (jeanlucf22) [inline tests/C-tests/threshold_matrix_typed.c:83]: This function needs to be called from all MPI tasks, not just task 0. This is probably why the MPI tests hang

Comment (jmohdyusof) [reply tests/C-tests/threshold_matrix_typed.c]: Thanks. Hopefully I caught all instances.

**Review by jeanlucf22 — APPROVED** [review]

**Review by nicolasbock — APPROVED** [review]
Please have a look at the merge conflicts.

**Comment by jmohdyusof** [comment]
The intel_offload versions are correct, I am not sure I have permissions to resolve them directly?

**Comment by nicolasbock** [comment]
You don't need to merge, you just have to
```
git rebase origin/master
```
I have resolved the conflicts and pushed a rebased branch. Let's see how CI thinks about that.

**Comment by jmohdyusof** [comment]
I edited the file and it triggered another CI run, so I am not sure that was needed.

**Comment by nicolasbock** [comment]
I did not get anything, but that you set up another CI is pretty cool!  
Would it be possible to get notifications and possibly log files 
somewhere public?

**Pull request merged**